cmd/libsnap-confine-private/mount-opt.c

```diff
@@ -115,3 +117,122 @@ const char *sc_mount_opt2str(unsigned long flags)
 	}
 	return buf;
 }
+
+static void sc_grow_string(char **s, const char *extra)
```

Comment:
When we're processing mount profiles we can see almost arbitrary mounts. I agree this is extra complexity but I think it is at some level unavoidable (and better than a static buffer one can overflow). One alternative would be to remove all logging entirely but I think this hurts debuggability -- often that list of mount operations is very useful when exploring unexpected behavior.